nixos/modules/services/home-automation/matter-server.nix

```diff
@@ -0,0 +1,123 @@
+{ lib
+, pkgs
+, config
+, ...
+}:
+
+with lib;
+
+let
+  cfg = config.services.matter-server;
+in
+
+{
+  meta.maintainers = with lib.maintainers; [ hexa ];
+
+  options.services.matter-server = with types; {
+    enable = mkEnableOption (lib.mdDoc "Matter-server");
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.python-matter-server;
+      defaultText = literalExpression "pkgs.python-matter-server";
+      description = "Package containing the matter-server executable.";
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 5580;
+      defaultText = "5580";
+      description = "Port to expose the matter-server service on.";
+    };
+
+    vendorId = mkOption {
+      type = types.ints.u16;
+      default = 4939; # Home Assistant
+      defaultText = "4939 (Home Assistant)";
+      description = "CSA Vendor ID to advertise to devices on the network.";
+    };
+
+    fabricId = mkOption {
+      type = types.int;
+      default = 1;
+      defaultText = "4939 (Home Assistant)";
+      description = "CSA Vendor ID to advertise to devices on the network.";
+    };
+
+    storagePath = mkOption {
+      type = types.nonEmptyStr;
+      default = "matter-server";
+      description = mdDoc "Directory name under `/var/lib/` to store persistent data. Specifying `\"matter-server\"` indicates the directory `/var/lib/matter-server`";
+    };
```

Comment:
```suggestion
```

Instead of exposing a storage-path, let's drop this option and people who want to store their state elsewhere can configure a bind mount on top of `/var/lib/matter-server`.

Systemd handles mounts on top of those directories extremely well, e.g. by setting up a dependency for the mount point on the unit.

Reply:
Good simplification, thanks for the suggestion.